### DISCUSSION TITLE
Disable invite and admin powers in direct chat

### DISCUSSION THREAD

**Discussion by scruloose** [element-hq/element-meta#537] [Ideas]
**Is your suggestion related to a problem? Please describe.**
It seems couterintuitive and confusing for  a direct chat to implement admin powers and the ability to invite other users:
* Kick, ban, mute, etc make no sense in a one-to-one chat: either party can just leave the conversation if they don't want to talk to the one and only other participant.
* New users are unlikely to expect a direct chat to have a setting for whether to allow newcomers to see the prior history, so it seems like this counter-intuitive behaviour could lead to unexpected privacy-breaching effects when *one* party assumes that a direct chat is just a direct chat but then the *other* party invites someone new in. Oops!
* For anyone considering Matrix/Riot as a replacement for SMS or Signal (shoutout to the Librem 5!), the confusing blurring-together of direct messaging and group chat rooms may be an obstacle.

**Describe the solution you'd like**
Honestly, I think the Matrix protocol should disable both inviting additional users and the whole admin powers framework in a direct chat between two users. But at least, can we throw a "Do you really want to convert this direct chat to a group room?" message when someone clicks the invite button in a direct chat? And include a warning about the prior history, if the current setting is to allow new users to see it?

**Describe alternatives you've considered**
I gather that the current behaviour is that the "direct chat" is converted into a "group chat room" when a third party joins. I have to admit that I don't see why that would ever be the expected or desirable way to do that, as opposed to creating a new room if I want to chat with more than one other user in one place.

**Additional context**
As a side-effect, I think disallowing inviting additional users to a group chat would take care of issue vector-im/element-web#9614 too.

**Comment by dud1337** [comment]
Disabling **invite** might be overkill. Otherwise, agreed. I've seen people have multiple matrix accounts and invite them all to the same "DM" with another user. Or occasionally adding a bot like giphy (not that I consider that secure). to DMs. In that case, a kick might be useful.

**Comment by tatarenko** [comment]
It would be great to have behavior similar to Skype: 
When you in DM trying to invite more people in it, app creates new room for 3 people, join original 2 persons automatically and switch focus to this new room with empty history, 3d one is invited to this room and they wait him there.
Current behavior leads to misunderstanding for our users - you have DM with like 1 year history with person, you invite colleague for some help, you find out yourself useless in this new conversation, you leave the room and delete it from you contact list and you lost those 1 year DM history but it is too complecated for you to understand why and when.
It would be great to have this as configurable option at least. We really miss it in our company.

**Comment by loucurapt** [comment]
If I understood correctly, if someone else is invited to a "direct message", that someone will be able to see the entire history of that "room" by default.
The default setting for "who can see the room history" is "Members only (since the point in time of selecting this option)".

That seems very wrong to me!
At the very least that setting should default to "Members only (since they were invited)" for direct messages, and should not be changeable.
(And the server/protocol allowing this also seems wrong to me.)

I think "direct messages" should be even more private/secure than private rooms.

**Comment by jxu** [comment]
Direct chat (under "People" tag) should not allow others to be invited. This defeats the purpose of a direct chat.

**Comment by xvsun** [comment]
We are currently testing the solution from the BWI called Bundesmessenger. They implemented a way so there is no invite button in DM in the webclient. Is this just a workaround hack or something that should be considered in Element? https://gitlab.opencode.de/bwi/bundesmessenger/clients/bundesmessenger-web

**Comment by cipherswami** [comment]
Absolutely yes, This invite and admin features makes no sense. I have recently started using Element and matrix, and I was pretty much overwhelmed by this new kind of things. 

I understand everything In matrix is a room. I was thinking like can we change this in matrix protocol itself, like classifying rooms under two categories DMs and Groups. Then clients can take care of, not using this unnecessary moderation for DM's.

**Reply by awtj8o81ryywg793** [reply]
If you’re looking to show or hide certain UI elements in different types of rooms, then you wouldn’t need to propose a change to the Matrix protocol itself, at least for the changes suggested in this discussion.

Compare the experience of navigating the same direct message in Element Classic and Element X.

**Comment by cipherswami** [comment]
And this event records are unnecessary in DMs as well. Just a request would be suffice, once it is accepted people can chat.

![img](https://github.com/user-attachments/assets/c066fc09-4bf4-464d-96ad-612fceb88de9)

**Reply by awtj8o81ryywg793** [reply]
Are you using Element Classic? You should know that Element Classic will, at some point, be sunset in favour of the next-generation Element X clients. The older clients are still supported (as of writing this reply), but you won’t be seeing any major changes going forward. Given that, I’d suggest using the current experience with Element X as your starting point.

In Element X, when you create a direct message, it indicates that encryption is enabled and whether you were invited to the room or you invited someone.

In regard to the status of the Element Classic clients, see the notice in each of their repositories:

(Edited to remove all links)

> This client is still supported and receives security updates but no new features or usability enhancements are made. It is recommended to use Element X that is the next-generation mobile app.

**Reply by cipherswami** [reply]
I just started with element and matrix. As of now I'm using both element classic and element x. As element x is still under development most of the features are not available yet.

And speaking generally, it is still need to be done on the web and desktop versions.

**Reply by awtj8o81ryywg793** [reply]
Quoted from the [previous reply](https://github.com/element-hq/element-meta/discussions/537#discussioncomment-14723991):

> And speaking generally, it is still need to be done on the web and desktop versions.

There is ongoing work to create an Element X Web/Desktop, not as a complete rewrite like we have seen with the mobile clients, but rather as incremental updates to the existing codebase. I'm assuming, though I might be wrong here, that as part of this process, the web and desktop clients will resemble the mobile clients more in this regard when it comes to the exposed functionality in rooms based on their type.

I would welcome a more consistent user interface, not just for rooms, but across the application.

**Reply by cipherswami** [reply]
Oh man why did they choose to rewrite it in kotlin and swift. For any major change they have to do it separately for 3 platforms including desktop.

Is electron not working properly for mobile versions?

**Reply by awtj8o81ryywg793** [reply]
Quoted from the [previous reply](https://github.com/element-hq/element-meta/discussions/537#discussioncomment-14724317):

> why did they choose to rewrite it in kotlin and swift

With Element X, they didn't choose a different programming language for the rewrite. Element Classic for iOS is written mostly in Swift, just like Element X for iOS. Element Classic for Android is written in Kotlin, just like Element X for Android.

As stated, Element Web/Desktop _hasn't_ been rewritten.

Quoted from the [previous reply](https://github.com/element-hq/element-meta/discussions/537#discussioncomment-14724317):

> For any major change they have to do it separately for 3 platforms including desktop.

This has always been the case, at least depending on the change. Both Element X mobile clients are built upon the Rust Matrix SDK, so they share the same foundation.

Quoted from the [previous reply](https://github.com/element-hq/element-meta/discussions/537#discussioncomment-14724317):

> electron not working properly for mobile versions?

Electron is used to build cross-platform desktop applications, not cross-platform mobile applications. I’m sure you could get Electron to work on mobile one way or another, but on its own, it’s desktop only.

**Reply by cipherswami** [reply]
oh man I thought. Classic was built using electron.
